azurerm/internal/services/web/app_service.go

```diff
@@ -91,6 +91,32 @@ func schemaAppServiceGoogleAuthSettings() *schema.Schema {
 	}
 }
 
+func schemaAppServiceGithubAuthSettings() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"client_id": {
+					Type:     schema.TypeString,
+					Required: true,
+				},
+				"client_secret": {
+					Type:      schema.TypeString,
+					Required:  true,
+					Sensitive: true,
+				},
+				"oauth_scopes": {
+					Type:     schema.TypeList,
+					Optional: true,
+					Elem:     &schema.Schema{Type: schema.TypeString},
```

Comment:
and can this be validated too?

Reply:
pulled this in favour of adding as a separate PR.